tracer/test/Datadog.Trace.Security.Unit.Tests/FuzzEncoder.cs

```diff
@@ -0,0 +1,193 @@
+// <copyright file="FuzzEncoder.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using Bogus;
+using Datadog.Trace.AppSec.Waf;
+using Datadog.Trace.AppSec.Waf.Initialization;
+using Datadog.Trace.AppSec.Waf.NativeBindings;
+using Datadog.Trace.Vendors.Newtonsoft.Json;
+using Datadog.Trace.Vendors.Newtonsoft.Json.Linq;
+using Xunit;
+
+namespace Datadog.Trace.Security.Unit.Tests
+{
+    public class FuzzEncoder
+    {
+        private static readonly JTokenType[] OpeningTypes = new[] { JTokenType.Array, JTokenType.Object };
+        private static readonly JTokenType[] TerminatingTypes = new[] { JTokenType.Integer, JTokenType.Float, JTokenType.String, JTokenType.Boolean, JTokenType.Null, JTokenType.Date, JTokenType.Guid, JTokenType.Uri, JTokenType.TimeSpan };
+        private readonly Random _rnd = new();
+        private readonly Faker faker = new("en");
+
+        [Fact]
+        [Trait("Category", "ArmUnsupported")]
+        public void LetsFuzz()
+        {
+            // if we don't throw any exceptions and generate a valid object the the test is successful
+
+            var libraryHandle = LibraryLoader.LoadAndGetHandle();
+            var wafNative = new WafNative(libraryHandle);
+            var encoder = new AppSec.Waf.Encoder(wafNative);
+
+            for (int i = 0; i < 100; i++)
+            {
+                var jsonReader = GenerateJson();
```

Comment:
This seems like a super useful util that will probably be handy to test on other integrations too 🙂 Maybe it should go in Datadog.Trace.Helpers?

Reply:
I did spend some time looking for something that did this already. I did think this was something that could be open sourced, but then I remembered we're already open source, so people can copy and paste it if they need it 😄 

I've moved to the helpers to make that easier, see dea63efa22f52e2aec7b97a34e6186345ac481eb.